### PR TITLE
Directive arguments in docs

### LINKED ISSUE #782: Some directives are missing their arguments in the docs
Some of the directives are missing some of their arguments in the docs.

This includes:

- `@all`
- `@first`
- `@paginate`

### PR DESCRIPTION
**Pull request by alberthaff**
Added definition for:

- [x] `@auth`
- [x] `@all`
- [x] `@belongsTo`
- [x] `@belongsToMany`
- [x] `@bcrypt`
- [x] `@broadcast`
- [x] `@builder`
- [x] `@cache`
- [x] `@cacheKey`
- [x] `@can`
- [x] `@complexity`
- [x] `@create`
- [x] `@delete`
- [x] `@deprecated`
- [x] `@field`
- [x] `@find`
- [x] `@first`
- [x] `@enum`
- [x] `@eq`
- [x] `@event`
- [x] `@globalId`
- [x] `@group`
- [x] `@hasMany`
- [x] `@hasOne`
- [x] `@in`
- [x] `@inject`
- [x] `@interface`
- [x] `@method`
- [x] `@middleware`
- [x] `@model`
- [x] `@neq`
- [x] `@node`
- [x] `@notIn`
- [x] `@orderBy`
- [x] `@paginate`
- [x] `@rename`
- [x] `@rules`
- [x] `@rulesForArray`
- [x] `@scalar`
- [x] `@search`
- [x] `@spread`
- [x] `@subscription`
- [x] `@trim`
- [x] `@union`
- [x] `@update`
- [x] `@where`
- [x] `@whereBetween`
- [x] `@whereNotBetween`
- [x] `@with`

**Related Issue/Intent**
Resolves #782 

**Changes**
This PR adds the definitions of directive's on the directives-page.

### CODE REVIEW

**Comment by spawnia** [comment]
That ToDo List 😃

**Comment by olivernybroe** [comment]
You are a hero! :D

File: docs/master/api-reference/directives.md [hunk 1]
```diff
@@ -128,6 +208,12 @@ type Mutation {
 }
 ```
 
+### Definition
+
+```graphql
+directive @bcrypt on FIELD_DEFINITION
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]: Actually, this rather applies to `ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION`

File: docs/master/api-reference/directives.md [hunk 2]
```diff
@@ -142,6 +228,24 @@ type Mutation {
 }
 ```
 
+### Definition
+
+```graphql
+directive @broadcast(
+  """
+  Name of the subscription.
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]:
```suggestion
  Name of the subscription that should be retriggered as a result of this operation.
```

File: docs/master/api-reference/directives.md [hunk 3]
```diff
@@ -142,6 +228,24 @@ type Mutation {
...
+  """
+  subscription: String
+  
+  """
+  Specify whether or not the job should be queued.
+  """
+  shouldQueue: Boolean
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]: Do we have a default here?

File: docs/master/api-reference/directives.md [hunk 4]
```diff
@@ -199,6 +303,24 @@ type Query {
 }
 ```
 
+### Definition
+
+```graphql
+directive @belongsToMany(
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]: ?

File: docs/master/api-reference/directives.md [hunk 5]
```diff
@@ -228,6 +350,12 @@ type GithubProfile {
 }
 ```
 
+### Definition
+
+```graphql
+directive @cacheKey on FIELD_DEFINITION
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]:
```suggestion
directive @cacheKey on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
```

File: docs/master/api-reference/directives.md (hunk 3 above)
Comment (alberthaff) [reply docs/master/api-reference/directives.md]: There is not a default value chosen in the `directiveArgValue()`, but the variable is passed on to `Subscription::broadcast()` which defaults `$shouldQueue = null`

File: docs/master/api-reference/directives.md [hunk 6]
```diff
@@ -250,6 +378,29 @@ class PostPolicy
...
+  
+  """
+  Additional arguments for policy check. 
+  """
+  args: [String!]
+  
+  """
+  Limit access to data, to currently authenticated user.
+  """
+  private: Boolean
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]: Where is the code for that?

Comment (alberthaff) [reply docs/master/api-reference/directives.md]: Typo - fixed.

File: docs/master/api-reference/directives.md [hunk 7]
```diff
@@ -371,6 +549,19 @@ type Mutation {
...
 ```
 
+### Definition
+
+```graphql
+directive @delete(  
+  """
+  Specify whether to use normal IDs or GlobalIDs.
+  """
+  globalId: Boolean
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]:
```suggestion
  globalId: Boolean = false
```

**Comment by alberthaff** [comment]
And then i managed to make a typo in that commit message... Good job.

File: docs/master/api-reference/directives.md (hunk 3 above)
Comment (spawnia) [reply docs/master/api-reference/directives.md]: Alright then, so it is a dynamic default value. Let's add it in the description.

File: docs/master/api-reference/directives.md [hunk 8]
```diff
@@ -300,6 +446,19 @@ type Query {
...
 ```
 
+### Definition
+
+```graphql
+directive @complexity(
+  """
+  Provide your own function to calculate complexity.
+  """
+  resolver: String
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]: Let's add a quick explanation of the `Class@method` format here and in other places where we require such method references.

File: docs/master/api-reference/directives.md [hunk 9]
```diff
@@ -538,6 +805,17 @@ type Mutation {
 }
 ```
 
+### Definition
+
+```graphql
+directive @event(  
+  """
+  Specify the event-class to dispatch.
```
Comment (spawnia) [inline docs/master/api-reference/directives.md]:
```suggestion
  Specify the fully qualified class name (FQCN) of the event dispatch.
```

**Comment by spawnia** [comment]
> And then i managed to make a typo in that commit message... Good job.

Hahaha i can not count the number of times i committed something like `Fix type`.

**Comment by alberthaff** [comment]
Not exactly sure what to do with `@enum`.

**Comment by spawnia** [comment]
> Not exactly sure what to do with `@enum`.

Done.

File: docs/master/api-reference/directives.md (hunk 8 above)
Comment (alberthaff) [reply docs/master/api-reference/directives.md]: I changed it to:

```
Provide your own function to calculate complexity.
Specify a fully qualified class name (FQCN) and method.
```

There is an example right below it. That should be sufficient. :-)

**Comment by chrissm79** [comment]
@alberthaff massive effort on this PR, thank you!! much more structured than before 👍

**Comment by alberthaff** [comment]
@spawnia @chrissm79 Thanks! :-)

**Pull request merged**